### DISCUSSION TITLE
Directing discussion

### DISCUSSION THREAD

**Discussion by jongfeel** [ThinkAboutSoftware/AcademicConference#30] [General]
## Problems

github pull request의 장점이자 단점입니다.
review 진행 시 원본 파일에 코멘트가 중간에 달리는 형태라
1:1로 리뷰 하기에는 상당히 좋은 방법인건 맞는데
1:4형태로 두번째, 세번째 학술회의를 진행하다 보니 아래와 같은 문제점이 있다는 걸 발견했습니다.

- 첫 리뷰를 진행한 사람은 원본 글에 대한 comment로 그나마 논의가 진행 됨
- 두번째, 세번째 부터는 첫 리뷰 진행한 comment에 사족 다는 정도로 진행함
- 그런 이유로 두번째, 세번째 리뷰 진행하는 사람은 새롭게 comment를 추가해 논의를 진행하기가 수월하지 않음

원본 글에 대한 논의는 없고 좋게 가자, 화이팅 하자, 반성 하자 이런 느낌의 코멘트가 주를 이루고 있습니다.

## To improve direction

그래서 다음과 같은 규칙으로 해봤으면 합니다.

- 두번째, 세번째 리뷰하는 사람도 되도록이면 첫번째 리뷰하는 사람의 코멘트를 따라 숟가락 얹기(?) 식으로 묻어가지 말고 원본 글에 대한 논의를 진행
- 공교롭게도 같은 line이 겹쳐서 첫번째 comment 단 사람의 글에 쓸 수 밖에 없는 상황이라면 다음 코멘트 앞에 `[New]`라고 붙이고 원본 글에 대한 논의라는 걸 표시하고 comment 추가
- `[New]`라는 원본 글에 대한 comment 추가 이후 마음껏(?) 다른 사람이 써 놓은 comment에 즐거운 동의글, 숟가락 얹기 글 달기

요렇게 해보려고 합니다.

다른 의견이 있다면 코멘트 달아 주시면 되고, 다음 학술 회의 회차 부터는 이 규칙을 적용할 예정입니다.

**Comment by jongfeel** [comment]
@yoonucho 
@hi-hyein 
@Joseph-Cha 

Notification commented

**Reply by Joseph-Cha** [reply]
저도 멘토님이 말씀해주신 부분은 지양하는 게 좋을 것 같습니다.

덧붙여서 같은 line의 리뷰 같은 경우에는 reply로 코멘트를 남기는 것이 아니라 처음 코멘트를 다는 것처럼 라인 선택 후 "+" 아이콘을 선택해서 코멘트를 다는 방식으로 한다면 각자 따로 따로 리뷰를 달 수 있습니다.

그래서 아래와 같이 하는 건 어떨까요?

- "+" 아이콘을 선택해서 코멘트를 다는 방식 : 원본 글에 대한 논의를 진행
- reply로 코멘트를 남기기 : 다른 사람이 남긴 코멘트에 덧붙여서 이야가 하고 싶을 때

**Reply by jongfeel** [reply]
아 저는 같은 line에 +로 추가 comment를 못단다고 생각했습니다.

그러면 이렇게 가죠.

- 되도록 원문 글에 대한 comment 추가
- 이미 comment 남긴 사람과 같은 line이라면 reply를 남기지 말고 +버튼을 이용해 comment 추가
  - 제가 제안했던 `[New]` 키워드 폐기
- 그 이후에 먼저 comment 남긴 사람에 대한 reply 작성

이렇게 하면 되겠네요.

+버튼이 뭔지 모르는 분들을 위해 알려드리면...

![image](https://user-images.githubusercontent.com/17442457/118589735-f15c2200-b7db-11eb-9171-8029458a72d9.png)

**Reply by jongfeel** [reply]
마지막에 혜인님이 좋은 방법을 알려 주셔서 아래의 최종 방법으로 적용해 보려고 합니다.
다음 부터 해보면 좋을 것 같아요.

- 원문 글에 대한 line별 comment 추가, 단 pending으로 추가만 해두고 approve는 하지 않음
- line comment를 다 작성한 사람은 pull request에 일반 comment로 "코멘트 추가 완료" 정도의 완료 메시지를 적음
- 마지막으로 완료했다는  comment가 추가 되면 그 이후 모두 시간 되는 대로 pending 되어 있던 review를 approve해서 완료.
- 다른 사람들이 단 comment에 추가 comment가 필요하면 추가로 작성
